### PR TITLE
Improve calculate go enrichment

### PR DESCRIPTION
**Pull request by jpquast**
Add new argument to `calculate_go_enrichment()` function.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # protti 0.7.0.9000
```
Comment (elena-krismer) [inline NEWS.md:2]: Shouldnt we release protti 0.8.0 0r 0.7.1? Since we are adding several new features https://semver.org/

File: R/calculate_go_enrichment.R [hunk 2]
```diff
@@ -55,24 +58,49 @@ go_enrichment <- function(...) {
...
+#' heatmap plot. Default is `mako_colours`.
+#' @param heatmap_fill_colour_rev a logical value that specifies if the provided colours in `heatmap_fill_colour`
+#' should be reversed in order. Default is `TRUE`.
 #' @param label a logical argument indicating whether labels should be added to the plot.
 #' Default is TRUE.
+#' @param enrichment_type a character argument that is either "all", "enriched" or "deenriched". This
+#' determines if the enrichment analysis should be performed in order to check for both enrichemnt and
+#' deenrichemnt or only one of the two. This affects the statistics performed and therefore also the displayed
+#' plot.
+#' @param min_n_detected_proteins_in_process is a numeric argument that specifies the minimal number
```
Comment (elena-krismer) [inline R/calculate_go_enrichment.R:85]: i would rephrase this argument slightly to make it more clear/easier to read: specifies the minimum number of detected proteins required for a GO term to be displayed in the plot. - or something like this

File: R/calculate_go_enrichment.R [hunk 3]
```diff
@@ -55,24 +58,49 @@ go_enrichment <- function(...) {
...
+#' detected in for example 1 protein. Even though these terms are sometimes significant, they are not
+#' really relevant.
 #' @param plot_cutoff a character value indicating if the plot should contain the top 10 most
 #' significant proteins (p-value or adjusted p-value), or if a significance cutoff should be used
 #' to determine the number of GO terms in the plot. This information should be provided with the
 #' type first followed by the threshold separated by a space. Example are
-#' \code{plot_cutoff = "adj_pval top10"}, \code{plot_cutoff = "pval 0.05"} or
-#' \code{plot_cutoff = "adj_pval 0.01"}. The threshold can be chosen freely. The default value is
-#' \code{"adj_pval top10"}.
+#' `plot_cutoff = "adj_pval top10"`, `plot_cutoff = "pval 0.05"` or
```
Comment (elena-krismer) [inline R/calculate_go_enrichment.R:95]: I would add a small test in the function as well to check the input from the user. Something like this:

```
if (!grepl("^(pval|adj_pval) (top10|\\d+(\\.\\d+)?)$", plot_cutoff)) {
    stop("Invalid format for plot_cutoff. Please provide the type (pval or adj_pval) followed by 'top10' or a numeric threshold within the range (0, 1]. Valid formats are for instance: 'adj_pval top10', 'pval 0.05'")
  }```

**Review by elena-krismer — APPROVED** [review]
Looks good! :)

File: NEWS.md (hunk 1 above)
Comment (jpquast) [reply NEWS.md:2]: Yes, I would make it protti 0.8.0 because we introduce some new features. During development I keep the version number like this to indicate that it is not yet a new release. Before releasing it I change it to the new release version number.

Comment (elena-krismer) [reply NEWS.md:2]: Ah okay nice!

File: R/calculate_go_enrichment.R (hunk 3 above)
Comment (jpquast) [reply R/calculate_go_enrichment.R:95]: Thanks! I added this but also modified the plot_cutoff so that "top" can actually be used with any number.
